### PR TITLE
Apply property shorthand syntax following ECMAScript 6 format

### PR DESCRIPTION
**Pull request by muxin**
#3504

From [previous PR](https://github.com/ampproject/amphtml/pull/3421#discussion-diff-65625388)
It might be better to apply [property shorthand syntax](http://es6-features.org/#PropertyShorthand) according to ES6 format throughout the code base for consistency.

For example, converting
```javascript
obj = { x: x, y: y };
```
to
```javascript
obj = { x, y };
```

However, if that's the convention we want, I find it not that intuitive when it comes to cases like the following:
```javascript
this.changeObservable_.fire({
  relayoutAll: relayoutAll,
  top: scrollTop,
  left: scrollLeft,
  width: size.width,
  height: size.height,
  velocity: velocity,
});
```
if I change the above block to
```javascript
this.changeObservable_.fire({
  relayoutAll,
  top: scrollTop,
  left: scrollLeft,
  width: size.width,
  height: size.height,
  velocity,
});
```
I think it introduced un-intuitiveness here.

What do you think about it? @cramforce @dvoytenko @mkhatib

### CODE REVIEW

**Comment by mkhatib** [comment]
Yeah, agreed mixed usage of the shorthand and non-shorthand in the same object literal might be very off-setting. One thing we can do to reduce that is to either:

1. Only use short-hand when all of the properties can be shorthanded OR

2. Allow mixing BUT put all short-handed properties at the top of the object literal, like the following.

```javascript
this.changeObservable_.fire({
  relayoutAll,
  velocity,
  top: scrollTop,
  left: scrollLeft,
  width: size.width,
  height: size.height,
});
```

**Comment by jridgewell** [comment]
I like option 2.

**Comment by cramforce** [comment]
Lets pick the one option our linter can already enforce.
@erwinmombay Is there something we can use?

**Comment by muxin** [comment]
I found a lint rule [here](http://eslint.org/docs/rules/object-shorthand). However it seems to be enforcing every instance no matter whether it mixes with non-shorthand properties or not.
On the eslint page, it said that 

> Others may find the terseness of the shorthand syntax harder to read and may not want to encourage it with this rule.

I don't have a strong opinion here. But enforcing the shorthand property syntax could be a pain for readability and also I am having some compiling errors from the closure compiler(looking into it). I wonder what do you guys think?

**Comment by cramforce** [comment]
I have little opinion. But not something worth spending more than a couple
hours on.

**Comment by erwinmombay** [comment]
yep @cramforce , pointed @muxin to eslint rules

**Comment by erwinmombay** [comment]
@muxin can you send me a screenshot of the errors for closure compiler? thanks

**Comment by muxin** [comment]
@erwinmombay This is the link to the failed travis build
https://travis-ci.org/ampproject/amphtml/builds/135740123

**Comment by cramforce** [comment]
I'm currently working on upgrading our compiler version to the latest. Lets
not worry about this error until after that has landed.

**Comment by muxin** [comment]
The closure compiler complains when I am changing
```javascript
export const Util = {
  filter: filter,
  guid: guid,
  log: log,
  make: make,
  set: set,
};
```
to
```javascript
export const Util = {filter, guid, log, make, set};
```
I don't know why that happens but my second commit makes the closure compiler work fine.

**Comment by muxin** [comment]
FYI. I made the changes in this pull request by searching `\b([a-zA-Z]+): \1\b` using regular expression, and then manually left out object that is a mix.

File: .eslintrc [hunk 1]
```diff
@@ -61,6 +61,7 @@
       "objectsInObjects": false,
       "arraysInObjects": false
     }],
+    "object-shorthand": [2, "properties"],
```
Comment (muxin) [inline .eslintrc]: This rule will make the linter complain every time we use a non-shorthand syntax. Shall I make the lint level `error` or `warning`? Do I have to separate this change to another PR?

File: extensions/amp-carousel/0.1/test/test-slides.js [hunk 2]
```diff
@@ -382,8 +382,8 @@ describe('Slides functional', () => {
         pos: 0.6,
         min: 0,
         max: 0,
-        prevTr: prevTr,
-        nextTr: nextTr,
+        prevTr,
```
Comment (erwinmombay) [inline extensions/amp-carousel/0.1/test/test-slides.js]: so we're ok with the shorthand being anywhere? i thought from the conversation here that they should always be on top

Comment (muxin) [inline extensions/amp-carousel/0.1/test/test-slides.js]: I'll add another commit to rearrange the orders.

File: .eslintrc (hunk 1 above)
Comment (muxin) [inline .eslintrc]: @erwinmombay What do you think about the error level? I found that travis would still fail if it sees lint warnings. So we might make it `error` level? Do you think I should make the lint rule a separate PR?

File: extensions/amp-carousel/0.1/test/test-slides.js (hunk 2 above)
Comment (erwinmombay) [inline extensions/amp-carousel/0.1/test/test-slides.js]: just want to clarify you are doing the work in this PR or another PR?

**Comment by erwinmombay** [comment]
@muxin LGTM, please make sure to fetch the latest from master and run the linter again before merging this

File: extensions/amp-carousel/0.1/test/test-slides.js (hunk 2 above)
Comment (muxin) [inline extensions/amp-carousel/0.1/test/test-slides.js]: In the same PR, commit coming

File: .eslintrc (hunk 1 above)
Comment (erwinmombay) [inline .eslintrc]: last conversation i think we had in channel, lets go with error. better for consistency. you can leave the change here just make sure to pull i the latest and run the linter again

Comment (muxin) [inline .eslintrc]: OK sure, thanks for the confirmation.

File: extensions/amp-carousel/0.1/test/test-slides.js (hunk 2 above)
Comment (muxin) [inline extensions/amp-carousel/0.1/test/test-slides.js]: On second thought, I think leaving the order as it is might be better. Sometimes we have similar objects of the same set of properties but different values like the following:
```javascript
object1 {
  x: x,
  y: bla,
  z: z,
}
object2 {
  x: x,
  y: bla,
  z: null,
}
```
I think converting it to
```javascript
object1 {
  x,
  y: bla,
  z,
}
object2 {
  x,
  y: bla,
  z: null,
}
```
makes more sense because object1 and object2 looks better if they have the same pattern.

@jridgewell @mkhatib @erwinmombay What do you think?

Comment (erwinmombay) [inline extensions/amp-carousel/0.1/test/test-slides.js]: honestly i don't have any strong opinion on this. since we can't enforce it right now anyways i do think we should just leave it as is

**Comment by cramforce** [comment]
👍
